### PR TITLE
Reskin

### PR DESCRIPTION
**Pull request by ohmoses**
(@eduardoboucas reviewed until: 3620f2cb958194bd763fec6e22d502ceb4c1b6cf)

### CODE REVIEW

File: app/components/RichEditor/RichEditor.jsx [hunk 1]
```diff
@@ -193,6 +194,9 @@ export default class RichEditor extends React.Component {
 
   handleChange({value}) {
     const {onChange} = this.props
+    const {isFocused} = value.selection
+
+    if (this.state.isFocused !== isFocused) this.setState({isFocused})
```
Comment (eduardoboucas) [inline app/components/RichEditor/RichEditor.jsx]:
```suggestion
    if (this.state.isFocused !== isFocused) {
      this.setState({isFocused})
    }
```

File: app/components/Table/TableRowCell.jsx [hunk 2]
```diff
@@ -33,7 +33,18 @@ export default class TableRowCell extends React.Component {
       this.props.select
     )
 
-    return <td className={cellStyle.getClasses()}>{this.renderChildren()}</td>
+    const {onHover} = this.props
+
+    return (
+      <td className={cellStyle.getClasses()}>
+        {React.Children.map(this.renderChildren(), child =>
+          // HTML elements have type 'string'; we only want to pass `onHover` to React Components.
```
Comment (eduardoboucas) [inline app/components/Table/TableRowCell.jsx]: I don't understand what this means. Could you rephrase?

File: app/edit-ui/src/util/classnames.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+export default function classnames(...args) {
```
Comment (eduardoboucas) [inline app/edit-ui/src/util/classnames.js]: It would be useful to have some comments explaining what this util is for and what it does exactly. Is it meant to live in tandem with or replace the existing `Style` class, and why?

File: app/edit-ui/src/components/TextInput/TextInput.jsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  const maybeResize = useCallback(() => {
+    if (autoresize && !resizable && ref.current) {
+      ref.current.style.height = 'auto'
+      ref.current.style.height = `${ref.current.scrollHeight + 2}px`
+    }
+  }, [autoresize, ref, resizable])
+
+  useEffect(maybeResize, [value])
+
+  // Cursor maintaining section
```
Comment (eduardoboucas) [inline app/edit-ui/src/components/TextInput/TextInput.jsx]: I really, really want to get to the bottom of why we need this in the first place.

File: test/api/cache/api/db.json [hunk 5]
```diff
@@ -1 +1,73 @@
...son","collections":[{"name":"cache","data":[],"idIndex":[],"binaryIndices":{},"constraints":null,"uniqueNames":[],"transforms":{},"objType":"cache","dirty":false,"cachedIndex":null,"cachedBinaryIndex":null,"cachedData":null,"adaptiveBinaryIndices":true,"transactional":false,"cloneObjects":false,"cloneMethod":"parse-stringify","asyncListeners":false,"disableMeta":false,"disableChangesApi":true,"disableDeltaChangesApi":true,"autoupdate":false,"serializableIndices":true,"ttl":null,"maxId":0,"DynamicViews":[],"events":{"insert":[],"update":[],"pre-insert":[],"pre-update":[],"close":[],"flushbuffer":[],"error":[],"delete":[null],"warning":[null]},"changes":[]}],"databaseVersion":1.5,"engineVersion":1.5,"autosave":true,"autosaveInterval":1000,"autosaveHandle":null,"throttledSaves":true,"options":{"autoload":true,"autosave":true,"autosaveInterval":1000,"serializationMethod":"normal","destructureDelimiter":"$<\n","recursiveWait":true,"recursiveWaitLimit":false,"recursiveWaitLimitDuration":2000,"started":1547565359365},"persistenceMethod":"fs","persistenceAdapter":null,"verbose":false,"events":{"init":[null],"loaded":[],"flushChanges":[],"close":[],"changes":[],"warning":[]},"ENV":"NODEJS"}
```
Comment (eduardoboucas) [inline test/api/cache/api/db.json]: We need Prettier to ignore this file.

File: app/edit-ui/src/components/TextInput/TextInput.jsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+      cursor.current = [selectionStart, selectionEnd]
+
+      props.onChange(e)
+    },
+    [props.onChange]
+  )
+
+  // Render
+
```
Comment (eduardoboucas) [inline app/edit-ui/src/components/TextInput/TextInput.jsx]: Nitpick: I know this isn't formally documented anywhere, but can we try to maintain the comment styles consistent with the rest of the codebase and other apps? Comments with ~80 max width, with a blank line before but not after, terminated with a period.

```suggestion
```

Comment (ohmoses) [reply app/edit-ui/src/components/TextInput/TextInput.jsx]: Hm, yes, I thought for these kinds of "section header" comments it would make more sense like this, but they work fine with the periods too

File: test/api/cache/api/db.json (hunk 5 above)
Comment (ohmoses) [reply test/api/cache/api/db.json]: Done

File: app/edit-ui/src/components/TextInput/TextInput.jsx (hunk 4 above)
Comment (ohmoses) [reply app/edit-ui/src/components/TextInput/TextInput.jsx]: I'll be glad to look into it! I did want to avoid this logic in the new TextInput component and fix it in Publish instead, but I wanted to get some visible results first, before you leave

File: app/edit-ui/src/util/classnames.js (hunk 3 above)
Comment (ohmoses) [reply app/edit-ui/src/util/classnames.js]: Oh, this is just a copypaste of classnames – https://www.npmjs.com/package/classnames (will add a comment to that effect).

I use it in the component library, because it's the standard way to do it (10k+ dependents on npm, 3M+ weekly downloads). It's so tiny and the API so stable that I didn't feel like installing it as a dependency, so I copied it instead (I rewrote it to ES6 code, but otherwise it's the same).

In Publish, I'd continue using Style just so that we don't introduce inconsistent ways of doing things. In new repos, I'd use classnames.

File: app/components/Table/TableRowCell.jsx (hunk 2 above)
Comment (ohmoses) [reply app/components/Table/TableRowCell.jsx]: Done

File: app/components/RichEditor/RichEditor.jsx (hunk 1 above)
Comment (ohmoses) [reply app/components/RichEditor/RichEditor.jsx]: Done

File: app/edit-ui/src/components/TextInput/TextInput.jsx (hunk 6 above)
Comment (ohmoses) [reply app/edit-ui/src/components/TextInput/TextInput.jsx]: Done

File: app/components/DocumentListToolbar/DocumentListToolbar.css [hunk 7]
```diff
@@ -1,91 +1,138 @@
...
-  }
+.page-button:hover:not(:disabled) {
+  color: var(--edit-color-text-2);
 }
 
-.selection-counter {
-  display: none;
-  padding-left: 8px;
-  visibility: hidden;
+#prev-icon,
```
Comment (eduardoboucas) [inline app/components/DocumentListToolbar/DocumentListToolbar.css]: Why use IDs?

File: app/components/DocumentListToolbar/DocumentListToolbar.jsx [hunk 8]
```diff
@@ -47,89 +48,123 @@ class DocumentListToolbar extends React.Component {
...
+    const {limit, offset, page, totalCount, totalPages} = metadata
+    const {goToPageValue} = this.state
 
-      result[page] = `Page ${page}`
+    const currentPageMin = (offset + 1).toLocaleString()
+    const currentPageMax = Math.min(offset + limit, totalCount).toLocaleString()
+    const numTotalDocuments = totalCount.toLocaleString()
+    const numSelectedDocuments = selectedDocuments.length.toLocaleString()
+
+    const pagesAroundCurrent = 2 // In each direction.
```
Comment (eduardoboucas) [inline app/components/DocumentListToolbar/DocumentListToolbar.jsx]: Should this be `PAGES_AROUND_CONTENT`, since it's a static value?

File: app/components/DocumentListToolbar/DocumentListToolbar.jsx [hunk 9]
```diff
@@ -138,56 +173,58 @@ class DocumentListToolbar extends React.Component {
...
-                      size="small"
-                      placeholder="Go to page"
-                      value={goToPageValue}
-                    />
-                  </form>
-                </span>
+        {totalCount > limit && (
+          <div className={styles.section}>
+            <div className={styles['page-numbers']}>
+              {totalPages > 9 ? (
```
Comment (eduardoboucas) [inline app/components/DocumentListToolbar/DocumentListToolbar.jsx]: Maybe make this `9` a variable and add a comment explaining where it comes from?

File: app/components/DocumentListToolbar/DocumentListToolbar.jsx [hunk 10]
```diff
@@ -47,89 +48,123 @@ class DocumentListToolbar extends React.Component {
...
+    const {goToPageValue} = this.state
 
-      result[page] = `Page ${page}`
+    const currentPageMin = (offset + 1).toLocaleString()
+    const currentPageMax = Math.min(offset + limit, totalCount).toLocaleString()
+    const numTotalDocuments = totalCount.toLocaleString()
+    const numSelectedDocuments = selectedDocuments.length.toLocaleString()
+
+    const pagesAroundCurrent = 2 // In each direction.
+    const ellipsisBefore = page > 2 + pagesAroundCurrent
```
Comment (eduardoboucas) [inline app/components/DocumentListToolbar/DocumentListToolbar.jsx]: I'd call these be `showEllipsisBefore` and `showEllipsisAfter`.

File: app/containers/CollectionNav/CollectionNav.jsx [hunk 11]
```diff
@@ -1,140 +1,81 @@
...
-              return {
-                id: slug,
-                label: settings.displayName || slug,
-                href: _publishLink,
-                isSelected
-              }
+    } = this.props
+
+    const items = buildGroupedMenuItems(api, collection)
+    const mobile = breakpoint === null
```
Comment (eduardoboucas) [inline app/containers/CollectionNav/CollectionNav.jsx]: `isMobile`?

Comment (ohmoses) [reply app/containers/CollectionNav/CollectionNav.jsx]: This is now deleted anyway

File: app/components/DocumentListToolbar/DocumentListToolbar.jsx (hunk 10 above)
Comment (ohmoses) [reply app/components/DocumentListToolbar/DocumentListToolbar.jsx]: Done

File: app/components/DocumentListToolbar/DocumentListToolbar.jsx (hunk 9 above)
Comment (ohmoses) [reply app/components/DocumentListToolbar/DocumentListToolbar.jsx]: Made it into a var; it doesn't really come from anywhere in particular, maybe just that 
```1 2 3 4 5 6 7 8 9```
is the same number of elements as 
```1 … 3 4 5 6 7 … 9```
so for this number of pages (and less) the ellipses don't save any space most of the time, but that's not important enough to put in a comment, I think

File: app/components/DocumentListToolbar/DocumentListToolbar.jsx (hunk 8 above)
Comment (ohmoses) [reply app/components/DocumentListToolbar/DocumentListToolbar.jsx]: Done

File: app/components/DocumentListToolbar/DocumentListToolbar.css (hunk 7 above)
Comment (ohmoses) [reply app/components/DocumentListToolbar/DocumentListToolbar.css]: Changed into classnames as per slack call

File: app/components/DocumentListToolbar/DocumentListToolbar.jsx [hunk 12]
```diff
@@ -13,6 +13,11 @@ function range(from, to) {
         .map((el, i) => from + i)
 }
 
+// If there are more pages than this, ellipses will be shown.
+const MAX_VISIBLE_PAGES = 9
+// Number of pages around the current one (in each direction) that will be visible.
```
Comment (eduardoboucas) [inline app/components/DocumentListToolbar/DocumentListToolbar.jsx]: Can you add a blank line, pretty please? 😇

File: app/components/FieldString/FieldString.css [hunk 13]
```diff
@@ -44,3 +14,27 @@
...
+
+.free-input-wrapper > * {
+  margin: 10px 0 0 10px;
+}
+
+.free-input-wrapper input {
+  flex: 1;
+}
+
+#open-icon {
```
Comment (eduardoboucas) [inline app/components/FieldString/FieldString.css]: Did you mean to leave this ID selector?

File: app/components/NavItem/NavItem.css [hunk 14]
```diff
@@ -3,27 +3,73 @@
...
 }
 
-.container .children {
-  display: none;
+.container.in-drawer .label {
+  margin: var(--edit-margin-sm) var(--edit-margin-md);
 }
 
-.container.expanded .children {
+#expand-icon {
```
Comment (eduardoboucas) [inline app/components/NavItem/NavItem.css]: Same question about ID selector.

File: app/components/Table/TableRowCell.jsx [hunk 15]
```diff
@@ -34,13 +34,12 @@ export default class TableRowCell extends React.Component {
...
-        {React.Children.map(this.renderChildren(), child =>
-          // If the child is a HTMLElement (type is a string), don't do anything.
-          // Otherwise the child is a React Component; pass the onHover callback.
-          typeof child.type === 'string'
-            ? child
-            : React.cloneElement(child, {onHover})
-        )}
+        {React.Children.map(this.renderChildren(), child => {
+          // Only pass onHover to field components.
+          return child.type.name.startsWith('Field')
```
Comment (eduardoboucas) [inline app/components/Table/TableRowCell.jsx]: Is this a good idea? 🤔 
.
.
.
.
.
.
.
.
.
.
.
.
.
.
.
.
.
.
(I don't have a better one.)

File: app/containers/Header/Header.css [hunk 16]
```diff
@@ -48,20 +95,19 @@
...
 }
 
-.overlay {
-  position: fixed;
-  left: 0;
-  right: 0;
-  top: 0;
-  bottom: 0;
-  z-index: var(--edit-z-index-overlay);
+#expand-icon {
```
Comment (eduardoboucas) [inline app/containers/Header/Header.css]: Same question.

File: app/containers/Header/Header.jsx [hunk 17]
```diff
@@ -11,55 +12,152 @@ class Header extends React.Component {
...
     return (
       <header className={styles.header}>
+        <div className={styles['collections-wrapper']} ref={this.outerRef}>
+          {collectionsInDrawer ? (
+            <div className={styles['drawer-wrapper']}>
+              <button
+                className={styles['drawer-toggle']}
+                onClick={this.toggleDrawer}
+              >
+                <i className="material-icons" id={styles['menu-icon']}>
```
Comment (eduardoboucas) [inline app/containers/Header/Header.jsx]: This will still be replaced by the new component for icons, right?

Comment (ohmoses) [reply app/containers/Header/Header.jsx]: Yes

File: app/components/FieldString/FieldString.css (hunk 13 above)
Comment (ohmoses) [reply app/components/FieldString/FieldString.css]: I didn't get to this yet, I'll do minor cleanups in the end (or in this case, when I switch them to the icon components)

File: app/components/DocumentListToolbar/DocumentListToolbar.jsx (hunk 12 above)
Comment (ohmoses) [reply app/components/DocumentListToolbar/DocumentListToolbar.jsx]: Done, will commit later

File: app/components/NavItem/NavItem.css (hunk 14 above)
Comment (ohmoses) [reply app/components/NavItem/NavItem.css]: See above

File: app/components/Table/TableRowCell.jsx (hunk 15 above)
Comment (ohmoses) [reply app/components/Table/TableRowCell.jsx]: I can try and come up with a better one when there's less work :man_shrugging: taking a note

File: app/containers/Header/Header.css (hunk 16 above)
Comment (ohmoses) [reply app/containers/Header/Header.css]: See above

File: app/components/DateTimePicker/DateTimePicker.jsx [hunk 18]
```diff
@@ -137,12 +109,18 @@ export default class DateTimePicker extends React.Component {
...
+      newDate.setUTCSeconds(overrides.seconds)
+    }
+
     return newDate
   }
 
   handleDatePick(date) {
     const {onChange} = this.props
 
+    console.log('!!!', date)
```
Comment (ohmoses) [inline app/components/DateTimePicker/DateTimePicker.jsx]: :thinking:

File: app/components/DateTimePicker/DateTimePicker.jsx [hunk 19]
```diff
@@ -203,23 +172,21 @@ export default class DateTimePicker extends React.Component {
       <div className={containerStyle.getClasses()}>
         <div className={styles.head}>
           <button
-            className={styles.arrow}
+            className={`${styles['page-icon']} ${styles['page-icon-prev']}`}
             onClick={this.handleMonthChange.bind(this, -1)}
```
Comment (ohmoses) [inline app/components/DateTimePicker/DateTimePicker.jsx]: Whenever I work on a file and I run into a function that is created and then garbage-collected with every re-render, I'd tidy that up into the constructor, i.e. something like
```
this.goToPrevMonth = this.handleMonthChange.bind(this, -1)
this.goToNextMonth = this.handleMonthChange.bind(this, 1)
```

File: app/components/DateTimePicker/DateTimePicker.jsx [hunk 20]
```diff
@@ -284,45 +239,53 @@ export default class DateTimePicker extends React.Component {
...
-          ref={element => {
-            this.hoursContainerRef = element
-          }}
-        >
-          {hours}
-        </ul>
-      </div>
+      <Select
+        dir="up"
+        onChange={e => this.handleDatePick(selectValues[e.target.value])}
```
Comment (ohmoses) [inline app/components/DateTimePicker/DateTimePicker.jsx]: I think I would strongly discourage creating *more* functions that are garbage collected on every render; it's just one small thing each time, but they add up

File: app/containers/SignIn/SignIn.jsx [hunk 21]
```diff
@@ -51,7 +50,7 @@ class SignIn extends React.Component {
 
         case 501:
           message =
-            'The API is running an earlier version than that required by this version of Publish'
+            'The API is running an earlier version than that required by this version of Edit'
```
Comment (ohmoses) [inline app/containers/SignIn/SignIn.jsx]: "…version of Edit **Interface**", I'd say, because the API is also Edit. Everything is Edit now :stuck_out_tongue:

File: app/components/DateTimePicker/DateTimePicker.jsx (hunk 18 above)
Comment (eduardoboucas) [reply app/components/DateTimePicker/DateTimePicker.jsx]: 🙈

File: app/components/DateTimePicker/DateTimePicker.jsx (hunk 19 above)
Comment (eduardoboucas) [reply app/components/DateTimePicker/DateTimePicker.jsx]: Done

File: app/components/DateTimePicker/DateTimePicker.jsx (hunk 20 above)
Comment (eduardoboucas) [reply app/components/DateTimePicker/DateTimePicker.jsx]: In this case it's a bit more involved, because the event handler needs access to the `selectValues` object we construct above.

File: app/containers/SignIn/SignIn.jsx (hunk 21 above)
Comment (eduardoboucas) [reply app/containers/SignIn/SignIn.jsx]: True. Although we need a global review of text and terminology, so I won't worry too much about it for now.

File: app/main.css [hunk 22]
```diff
@@ -60,3 +60,13 @@ select {
   --theme-font-size-large: 20px;
   --theme-toolbar-height: 63px;
 }
+
+:global #app {
+  height: 100%;
+  display: flex;
```
Comment (eduardoboucas) [inline app/main.css]: Why do we need to use flexbox on the root element? [This](https://jakearchibald.com/2014/dont-use-flexbox-for-page-layout/) keeps popping in my head.

File: app/components/DateTimePicker/DateTimePicker.jsx (hunk 20 above)
Comment (ohmoses) [reply app/components/DateTimePicker/DateTimePicker.jsx]: Ah, of course, I'm an idiot

File: app/main.css (hunk 22 above)
Comment (ohmoses) [reply app/main.css]: so that the `<main>` can get `flex: 1` and stretch vertically to the bottom of the screen (I'm happy to have alternative solutions)
